tests/cpp_unit_tests/test_distribution.cpp

```diff
@@ -0,0 +1,63 @@
+#include "openmc/distribution.h"
+#include "openmc/random_lcg.h"
+#include <catch2/catch_test_macros.hpp>
+#include <pugixml.hpp>
+
+TEST_CASE("Test alias method sampling of a discrete distribution")
+{
+  int n_samples = 1e6;
```

Comment:
Since this value is used to initialize the size of the array, it needs to be guaranteed to be a compile-time constant:
```suggestion
  constexpr int n_samples = 1000000;
```